pkg/services/libraryelements/database.go

```diff
@@ -441,6 +441,42 @@ func (l *LibraryElementService) getAllLibraryElements(c context.Context, signedI
 	if folderFilter.parseError != nil {
 		return model.LibraryElementSearchResult{}, folderFilter.parseError
 	}
+
+	var foldersWithMatchingTitles []string
+	if len(strings.TrimSpace(query.SearchString)) > 0 {
+		searchQuery := folder.SearchFoldersQuery{
+			OrgID:        signedInUser.GetOrgID(),
+			Title:        query.SearchString,
+			SignedInUser: signedInUser,
+		}
+
+		folderHits, err := l.folderService.SearchFolders(c, searchQuery)
+		if err != nil {
+			if strings.Contains(err.Error(), "cannot be called on the legacy folder service") {
+				fs, err := l.folderService.GetFolders(c, folder.GetFoldersQuery{
+					OrgID:        signedInUser.GetOrgID(),
+					SignedInUser: signedInUser,
+				})
+				if err != nil {
+					return model.LibraryElementSearchResult{}, err
+				}
+				foldersWithMatchingTitles = make([]string, 0, len(fs))
+				for _, f := range fs {
+					if strings.Contains(strings.ToLower(f.Title), strings.ToLower(query.SearchString)) {
+						foldersWithMatchingTitles = append(foldersWithMatchingTitles, f.UID)
+					}
+				}
+			} else {
+				return model.LibraryElementSearchResult{}, err
+			}
```

Comment:
```suggestion
		if err != nil && !strings.Contains(err.Error(), "cannot be called on the legacy folder service") {
		   return model.LibraryElementSearchResult{}, err
		}
		fs, err := l.folderService.GetFolders(c, folder.GetFoldersQuery{
			OrgID:        signedInUser.GetOrgID(),
			SignedInUser: signedInUser,
		})
		if err != nil {
			return model.LibraryElementSearchResult{}, err
		}
		foldersWithMatchingTitles = make([]string, 0, len(fs))
		for _, f := range fs {
			if strings.Contains(strings.ToLower(f.Title), strings.ToLower(query.SearchString)) {
				foldersWithMatchingTitles = append(foldersWithMatchingTitles, f.UID)
			}
		}
```

nit: About Go, we try to return early if there is an error. In this case, I would add the nested condition and return the errro when needed, and then, the happy path without nesting if/else.